### PR TITLE
 Reimplement #469 to add ROOT version dependency

### PR DESCRIPTION
**Pull request by syamasak**
See #471 for more information.

### CODE REVIEW

**Comment by ktf** [comment]
+test

**Comment by mkrzewic** [comment]
Hi, it's not clear to me why this is needed - does it fix a problem? If yes - when was it introduced? Is there a JIRA ticket related to it?

**Comment by ktf** [comment]
See #469. A few people actually complained about it on the mailing list. As best practice we should always have the closure of all dependencies we need.

**Comment by syamasak** [comment]
Dear @mkrzewic,

I am sorry for making you confused and lack of explanation.
The build error is happen because I tried to make AliRoot build pass on clean-installed macOS high sierra without disabling SIP. Previously this security feature has blocked the build of fastjet, but the pending PR ( https://github.com/alisw/alibuild/pull/495 ) should fix this issue. After successful build of fastjet, unfortunatelly stucked at #469.
There is a JIRA ticket (ALIROOT-7574) for related but different problem. It says even if AliRoot build itself is ok but the test always fails. This means AliPhysics is not buildable in some environment. It's good chance to fix all linkage issues around here, so this PR.
I'll create a JIRA ticket related to a PR for the next time.

Best, Shuto

**Comment by syamasak** [comment]
Any progress?

**Comment by rbertens** [comment]
Hi all, naive question: should this PR fix build issues reported in #469 ? If so, what's the status? 
Thanks,
Redmer

File: HLT/BASE/CMakeLists.txt [hunk 1]
```diff
@@ -149,6 +149,13 @@ get_directory_property(incdirs INCLUDE_DIRECTORIES)
 generate_dictionary("${MODULE}" "${MODULE}LinkDef.h" "${HDRS}" "${incdirs}")
 
 set(ROOT_DEPENDENCIES Core Geom Graf Hist MathCore Net RIO Tree XMLParser)
+
+if(${ROOT_VERSION_MAJOR} LESS 6)
+	list(APPEND ROOT_DEPENDENCIES Cint)
+else()
+	list(APPEND ROOT_DEPENDENCIES Cling)
+endif(${ROOT_VERSION_MAJOR} LESS 6)
```
Comment (dberzano) [inline HLT/BASE/CMakeLists.txt]: Can you turn it to a simple `endif()` please? Otherwise it looks good to me.

Comment (dberzano) [reply HLT/BASE/CMakeLists.txt]: Also, use spaces not tabs.

Comment (syamasak) [reply HLT/BASE/CMakeLists.txt]: Thanks, fixed!

**Comment by dberzano** [comment]
+1

**Pull request merged**